### PR TITLE
Document required PR lables

### LINKED ISSUE #343: Trigger tasks based on label(s)
As we're scaling up our CI usage, we frequently see contributions that take several commits to get into a good state. It'd be really nice if we could conditionally run jobs based on labels on 
the issue.

The things I'm concerned about are:

- Contributors often need multiple commits to address review comments.  
- We want to use compute credits but also want to avoid spending compute time on experimentation
- We want to avoid the CI system getting thrashed by an unwitting contributor (or, worse, an intentionally malicious contributor).

For example, a workflow that seems good would be something like this:

1. Contributor opens PR. A subset of CI jobs run (code linting/analysis, a small-ish set of smoke tests , ideally based on the files that have changed in the patch).
2. Contributor makes changes to the patch in response to review or perhaps in response to the smoke tests being run.
3. Someone with permissions to add labels eventually adds label to the issue that asks for all tests to be run. This is done when the patch is deemed as being (closer to) ready for submission, and is already passing the smoke tests. This triggers all presubmit tests and allows them to run with compute credits.
4. Additional commits from that point on run all presubmit tests.

### PR DESCRIPTION
**Pull request by fkorotkov**
Fixes #343

### CODE REVIEW

File: docs/guide/writing-tasks.md [hunk 1]
```diff
@@ -585,6 +585,25 @@ task:
...
+tests on every [supported platform](supported-computing-services.md). Simply use `required_pr_labels` field to specify
+a list of labels a PR requires to have in order to trigger a task. Here is a simple example of `.cirrus.yml` config
+that automatically runs a linting tool but requires `initial-review` label being presented in order to run tests:
+
+```yaml
+lint_task:
+  # ...
+
+test_task:
+  required_pr_labels: initial-review
```
Comment (dnfield) [inline docs/guide/writing-tasks.md]: Is it possible to have multiple labels, either inclusive or exclusive?

e.g. only run this task if all labels, or only run this task if at least one of the following labels?

Not sure if that's really needed, more curious

Comment (fkorotkov) [reply docs/guide/writing-tasks.md]: You can specify multiple labels and they all should be presented on the PR. For example:

```yaml
build_task:
  required_pr_labels:
    - cla-signed
    - initial-review

task:
  name: Deploy Staging
  depends_on: build
  required_pr_labels:
    - cla-signed
    - initial-review
    - ready-for-staging
```

Comment (dnfield) [reply docs/guide/writing-tasks.md]: Ahh ok - that would be helpful as an example in the doc

**Comment by dnfield** [comment]
This is awesome! thanks!

**Pull request merged**